### PR TITLE
RSDK-7307: Update Status for Board in Python SDK

### PR DESCRIPTION
**Pull request by martha-johnston**

### CODE REVIEW

**Comment by njooma** [comment]
Looks reasonable for now, but should be considered a draft until https://github.com/viamrobotics/api/pull/479 is merged

**Comment by martha-johnston** [comment]
marking as ready for review so that this is hopefully ready to merge once the rdk changes get approved, but I won't merge until api is merged

**Review by njooma — APPROVED** [review]

**Review by purplenicole730 — APPROVED** [review]

**Comment by martha-johnston** [comment]
@njooma @purplenicole730 do either of you know why these test might be failing? it seems unrelated to my change so wondering if you've seen it before.

**Comment by njooma** [comment]
> @njooma @purplenicole730 do either of you know why these test might be failing? it seems unrelated to my change so wondering if you've seen it before.

Not your fault, but you have to add an unimplemented function in [RobotService](https://github.com/viamrobotics/viam-python-sdk/blob/5736d37d37ba107400c650b3e502bb97b572831a/src/viam/robot/service.py). The function is `RestartModule`

File: src/viam/components/board/client.py
Comment (martha-johnston) [inline src/viam/components/board/client.py:171]: okay I was having a test issue and couldn't figure out how to resolve it without making these not optional. how much of a problem is that/does anyone know a better resolution?

**Pull request merged**
